scripts/delete_from_repository.py

```diff
@@ -0,0 +1,151 @@
+import argparse
+import logging
+import os
+import pprint
+import sys
+import time
+from urllib.parse import urljoin
+
+import requests
+from requests.auth import HTTPBasicAuth
+from sqlalchemy.future import select
+
+sys.path.append(os.path.dirname(os.path.dirname(__file__)))
+
+from alws import models
+from alws.config import settings
+from alws.database import SyncSession
+
+
+PROG_NAME = 'packages-repo-deleter'
+
+
+class RepositoryNotFound(ValueError):
+    pass
+
+
+def parse_args(args):
+    parser = argparse.ArgumentParser('packages-repo-deleter')
+    parser.add_argument('-r', '--repository-name', required=True, type=str)
+    parser.add_argument('-a', '--architecture', required=True, type=str)
+    parser.add_argument('-p', '--packages', nargs='+', required=True, type=str)
+    parser.add_argument('-v', '--verbose', action='store_true', default=False)
+    return parser.parse_args(args)
+
+
+def get_repository(repo_name: str, arch: str) -> models.Repository:
+    with SyncSession() as session:
+        repo = (session.execute(select(models.Repository)
+                                .where(models.Repository.name == repo_name,
+                                       models.Repository.arch == arch))
+                .scalars().first())
+        if not repo:
+            raise RepositoryNotFound(f'Repository {repo_name} is not found')
+        return repo
```

Comment:
How will package maintainers get access to our database?
IMHO, it's a lot of unneeded dependencies for this task

Reply:
Basically, it's a lot easier to get repo from pulp just by name: https://docs.pulpproject.org/pulp_rpm/restapi.html#operation/repositories_rpm_rpm_list
Also, from pulp response (autopublish) you can decide should or shouldn't create publication manually, it's a lot safer / require a lot less dependencies / can be runned on any device any time

Reply:
Do you mean we need to rely only on communication with pulp?

Reply:
Yes, this is how I did it previously, I can share my work with you
My script rely only on pulp responses and is safe to run from any place any time